### ISSUE TITLE
cross-compile fails at commit f33d152400c5a8acd9b8b00efd91a59328388d9a

### ISSUE THREAD

**Issue by hydra3333** [KhronosGroup/SPIRV-Tools#5001]
Hello. It used to cross-compile under mingw64/ubuntu with target windows.
Now it fails on and after commit f33d152400c5a8acd9b8b00efd91a59328388d9a

Log:

```
[13:42:50][DEBUG] ninja -j 6 
[13:42:50][DEBUG] Running 'ninja -j 6 ' in '/home/u/Desktop/_working/workdir/x86_64/spirv-tools/_build'
[0/1] Re-running CMake...
-- No build type selected, default to Debug
-- Configuring done
-- Generating done
-- Build files have been written to: /home/u/Desktop/_working/workdir/x86_64/spirv-tools/_build
[1/314] cd /home/u/Desktop/_working/workdir/x86_64/spirv-tools/_build && /usr/bin/cmake -DCHANGES_FILE=/home/u/Desktop/_working/workdir/x86_64/spirv-tools/CHANGES -DTEMPLATE_FILE=/home/u/Desktop/_working/workdir/x86_64/spirv-tools/cmake/SPIRV-Tools.pc.in -DOUT_FILE=/home/u/Desktop/_working/workdir/x86_64/spirv-tools/_build/SPIRV-Tools.pc -DCMAKE_INSTALL_PREFIX=/home/u/Desktop/_working/workdir/toolchain/x86_64-w64-mingw32/x86_64-w64-mingw32 -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_INSTALL_INCLUDEDIR=include -DSPIRV_LIBRARIES=-lSPIRV-Tools-opt\ -lSPIRV-Tools\ -lSPIRV-Tools-link -P /home/u/Desktop/_working/workdir/x86_64/spirv-tools/cmake/write_pkg_config.cmake
[2/314] cd /home/u/Desktop/_working/workdir/x86_64/spirv-tools/_build && /usr/bin/cmake -DCHANGES_FILE=/home/u/Desktop/_working/workdir/x86_64/spirv-tools/CHANGES -DTEMPLATE_FILE=/home/u/Desktop/_working/workdir/x86_64/spirv-tools/cmake/SPIRV-Tools-shared.pc.in -DOUT_FILE=/home/u/Desktop/_working/workdir/x86_64/spirv-tools/_build/SPIRV-Tools-shared.pc -DCMAKE_INSTALL_PREFIX=/home/u/Desktop/_working/workdir/toolchain/x86_64-w64-mingw32/x86_64-w64-mingw32 -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_INSTALL_INCLUDEDIR=include -DSPIRV_SHARED_LIBRARIES=-lSPIRV-Tools-shared -P /home/u/Desktop/_working/workdir/x86_64/spirv-tools/cmake/write_pkg_config.cmake
[3/314] Generate extended instruction tables for spv-amd-shader-explicit-vertex-parameter.
[4/314] Generate extended instruction tables for spv-amd-shader-trinary-minmax.
[5/314] Generate extended instruction tables for spv-amd-gcn-shader.
[6/314] Generate extended instruction tables for spv-amd-shader-ballot.
[7/314] Generate extended instruction tables for debuginfo.
[8/314] Generate info tables for SPIR-V vunified1 core instructions and operands.
[9/314] Generate language specific header for DebugInfo.
[10/314] Generate extended instruction tables for nonsemantic.shader.debuginfo.100.
[11/314] Generate language specific header for OpenCLDebugInfo100.
[12/314] Generate extended instruction tables for opencl.debuginfo.100.
[13/314] Generate extended instruction tables for nonsemantic.clspvreflection.
[14/314] Generate language specific header for NonSemanticShaderDebugInfo100.
[15/314] Generate tables based on the SPIR-V XML registry.
[16/314] Update build-version.inc in the SPIRV-Tools build directory (if necessary).
[17/314] Generate enum-string mapping for SPIR-V vunified1.
[18/314] Generate info tables for GLSL extended instructions and operands vunified1.
[19/314] Generate info tables for OpenCL extended instructions and operands vunified1.
[20/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/util/bit_vector.cpp.obj
[21/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/util/string_utils.cpp.obj
[22/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/assembly_grammar.cpp.obj
[23/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/diagnostic.cpp.obj
[24/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/ext_inst.cpp.obj
[25/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/enum_string_mapping.cpp.obj
[26/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/extensions.cpp.obj
[27/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/libspirv.cpp.obj
[28/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/disassemble.cpp.obj
[29/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/util/parse_number.cpp.obj
[30/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/opcode.cpp.obj
[31/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/software_version.cpp.obj
[32/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/spirv_endian.cpp.obj
[33/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/spirv_fuzzer_options.cpp.obj
[34/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/parsed_operand.cpp.obj
[35/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/binary.cpp.obj
[36/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/spirv_optimizer_options.cpp.obj
[37/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/spirv_reducer_options.cpp.obj
[38/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/operand.cpp.obj
[39/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/spirv_validator_options.cpp.obj
[40/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/name_mapper.cpp.obj
[41/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/spirv_target_env.cpp.obj
[42/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/table.cpp.obj
[43/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/print.cpp.obj
[44/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_adjacency.cpp.obj
[45/314] Building CXX object source/CMakeFiles/SPIRV-Tools-static.dir/text.cpp.obj
[46/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/text_handler.cpp.obj
[47/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/text.cpp.obj
[48/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_barriers.cpp.obj
[49/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_annotation.cpp.obj
[50/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_arithmetics.cpp.obj
[51/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate.cpp.obj
[52/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_atomics.cpp.obj
[53/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_bitwise.cpp.obj
[54/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_capability.cpp.obj
[55/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_composites.cpp.obj
[56/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_constants.cpp.obj
[57/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_conversion.cpp.obj
[58/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_debug.cpp.obj
[59/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_derivatives.cpp.obj
[60/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_execution_limitations.cpp.obj
[61/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_cfg.cpp.obj
[62/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_function.cpp.obj
[63/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_id.cpp.obj
[64/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_decorations.cpp.obj
[65/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_image.cpp.obj
[66/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_interfaces.cpp.obj
[67/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_instruction.cpp.obj
[68/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_layout.cpp.obj
[69/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_extensions.cpp.obj
[70/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_builtins.cpp.obj
[71/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_literals.cpp.obj
[72/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_logicals.cpp.obj
[73/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_memory_semantics.cpp.obj
[74/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_mesh_shading.cpp.obj
[75/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_misc.cpp.obj
[76/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_non_uniform.cpp.obj
[77/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_primitives.cpp.obj
[78/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_memory.cpp.obj
[79/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_ray_query.cpp.obj
[80/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_mode_setting.cpp.obj
[81/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_ray_tracing_reorder.cpp.obj
FAILED: source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_ray_tracing_reorder.cpp.obj 
/home/u/Desktop/_working/workdir/toolchain/x86_64-w64-mingw32/bin/x86_64-w64-mingw32-g++ --sysroot=/home/u/Desktop/_working/workdir/toolchain/x86_64-w64-mingw32 -DSPIRV_COLOR_TERMINAL -DSPIRV_TOOLS_IMPLEMENTATION -DSPIRV_TOOLS_SHAREDLIB -DSPIRV_Tools_shared_EXPORTS -DSPIRV_WINDOWS -I/home/u/Desktop/_working/workdir/x86_64/spirv-tools -I/home/u/Desktop/_working/workdir/x86_64/spirv-tools/include -I/home/u/Desktop/_working/workdir/x86_64/spirv-tools/_build -I/home/u/Desktop/_working/workdir/x86_64/spirv-tools/external/spirv-headers/include -O3  -fstack-protector-all  -D_FORTIFY_SOURCE=2 -g -fvisibility=hidden -DSPIRV_CHECK_CONTEXT -Wall -Wextra -Wnon-virtual-dtor -Wno-missing-field-initializers -Werror -std=c++11 -fno-exceptions -Wno-long-long -Wshadow -Wundef -Wconversion -Wno-sign-conversion -std=gnu++11 -MD -MT source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_ray_tracing_reorder.cpp.obj -MF source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_ray_tracing_reorder.cpp.obj.d -o source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_ray_tracing_reorder.cpp.obj -c /home/u/Desktop/_working/workdir/x86_64/spirv-tools/source/val/validate_ray_tracing_reorder.cpp
/home/u/Desktop/_working/workdir/x86_64/spirv-tools/source/val/validate_ray_tracing_reorder.cpp:25:49: error: 'numeric_limits' is not a member of 'std'
   25 | static const uint32_t KRayParamInvalidId = std::numeric_limits<uint32_t>::max();
      |                                                 ^~~~~~~~~~~~~~
/home/u/Desktop/_working/workdir/x86_64/spirv-tools/source/val/validate_ray_tracing_reorder.cpp:25:72: error: expected primary-expression before '>' token
   25 | static const uint32_t KRayParamInvalidId = std::numeric_limits<uint32_t>::max();
      |                                                                        ^
/home/u/Desktop/_working/workdir/x86_64/spirv-tools/source/val/validate_ray_tracing_reorder.cpp:25:75: error: '::max' has not been declared; did you mean 'std::max'?
   25 | static const uint32_t KRayParamInvalidId = std::numeric_limits<uint32_t>::max();
      |                                                                           ^~~
      |                                                                           std::max
In file included from /home/u/Desktop/_working/workdir/toolchain/x86_64-w64-mingw32/include/c++/12.2.0/algorithm:61,
                 from /home/u/Desktop/_working/workdir/x86_64/spirv-tools/source/val/validation_state.h:18,
                 from /home/u/Desktop/_working/workdir/x86_64/spirv-tools/source/val/validate_ray_tracing_reorder.cpp:20:
/home/u/Desktop/_working/workdir/toolchain/x86_64-w64-mingw32/include/c++/12.2.0/bits/stl_algo.h:5756:5: note: 'std::max' declared here
 5756 |     max(initializer_list<_Tp> __l, _Compare __comp)
      |     ^~~
[82/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/basic_block.cpp.obj
[83/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_ray_tracing.cpp.obj
[84/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_small_type_uses.cpp.obj
[85/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_scopes.cpp.obj
[86/314] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_type.cpp.obj
ninja: build stopped: subcommand failed.

```

**Comment by alan-baker** [comment]
Fixed by #5002

**Comment by hydra3333** [comment]
Thanks
